### PR TITLE
Migrate Focal Modulation Network tutorial to Keras 3

### PR DESCRIPTION
**Pull request by LakshmiKalaKadali**
This PR migrates the Focal Modulation Network image classification tutorial from Keras 2 to Keras 3. The implementation is now fully backend-agnostic, allowing it to run seamlessly on JAX, PyTorch, and TensorFlow ([notebook](https://colab.research.google.com/gist/LakshmiKalaKadali/0d1a6da651257412e683470772cf11c4/focal-modulation-a-replacement-for-self-attention.ipynb#scrollTo=kpNzRCj_tuxO)).
The changes incorporated:

Data Pipeline: Migrated from tf.data.Dataset to keras.utils.PyDataset. This ensures that data loading remains consistent and performant across different backends without encountering symbolic tensor errors.
Integrated Preprocessing: Moved Rescaling, Resizing, and data augmentation layers (RandomFlip, RandomCrop) directly into the model's call method. This is the recommended Keras 3 pattern to ensure preprocessing is part of the serialized model and runs on the target device.
Numerical Stability Fixes:
Added ops.softmax to the gating mechanism in the FocalModulationLayer to prevent value explosion (a common issue with hierarchical convolutions).
Enabled clipnorm=1.0 in the AdamW optimizer to stabilize training.

### CODE REVIEW

File: examples/vision/focal_modulation_network.py [hunk 1]
```diff
@@ -566,68 +502,41 @@ class FocalModulationBlock(layers.Layer):
...
-        # FFN
-        x = shortcut + x
-        x = x + self.mlp(self.norm(x))
-        return x
+        res = x
+        x = ops.reshape(x, (-1, height, width, channels))
+        x = self.modulation(x, training=training)
+        x = ops.reshape(x, (-1, height * width, channels))
+        x = res + x
+        return x + self.mlp(self.norm2(x), training=training)
```
[the remark this answers is not in the document]
Comment (LakshmiKalaKadali) [reply examples/vision/focal_modulation_network.py:538]: Without the Block's skip connections (res + x), the network loses its residual paths. This would cause vanishing gradients, making the model numerically unstable and preventing convergence

File: examples/vision/focal_modulation_network.py [hunk 2]
```diff
@@ -93,24 +97,23 @@
...
 LEARNING_RATE = 1e-4
 WEIGHT_DECAY = 1e-4
-
-# TRAINING
-EPOCHS = 25
+EPOCHS = 20
 
 """
-## Load and process the CIFAR-10 dataset
+## Data Loading with PyDataset
```
Comment (sachinprasadhs) [inline examples/vision/focal_modulation_network.py:112]: Add a line separator after this

Comment (sachinprasadhs) [reply examples/vision/focal_modulation_network.py:112]: Retain the same heading "Load and process the CIFAR-10 dataset" since you are explaining in the description below

File: examples/vision/focal_modulation_network.py [hunk 3]
```diff
@@ -311,10 +281,9 @@ def MLP(
 ):
```
Comment (sachinprasadhs) [inline examples/vision/focal_modulation_network.py]: Can you remove type annotation in this MLP class as well, so that it remains consistent with your change and you can remove the import from typing import Optional

File: examples/vision/focal_modulation_network.py (hunk 2 above)
Comment (LakshmiKalaKadali) [reply examples/vision/focal_modulation_network.py:112]: Done suggested changes

File: examples/vision/focal_modulation_network.py (hunk 3 above)
Comment (LakshmiKalaKadali) [reply examples/vision/focal_modulation_network.py]: Yes removed both. Thanks

File: examples/vision/focal_modulation_network.py [hunk 4]
```diff
@@ -71,20 +72,24 @@
...
 import numpy as np
-import tensorflow as tf
-from tensorflow import keras
-from tensorflow.keras import layers
-from tensorflow.keras.optimizers.experimental import AdamW
-from typing import Optional, Tuple, List
+import keras
+from keras import layers
+from keras import ops
+from typing import Optional
```
Comment (sachinprasadhs) [inline examples/vision/focal_modulation_network.py]: You can remove the optional as well.

File: examples/vision/focal_modulation_network.py [hunk 5]
```diff
@@ -93,24 +98,23 @@
...
+EPOCHS = 20
 
 """
-## Load and process the CIFAR-10 dataset
+## Data Loading with PyDataset
+Keras 3 introduces PyDataset as a standardized way to handle data.
+It works identically across all backends and avoids the "Symbolic Tensor" issues often found
+when using tf.data with JAX or PyTorch.
+
 """
```
Comment (sachinprasadhs) [inline examples/vision/focal_modulation_network.py:117]: Extra line space after heading "## line"

before """ there should not be any extra line space.

**Review by sachinprasadhs — APPROVED** [review]
LGTM, Thanks.
Can you regenerate .md and .ipynb with the latest changes of .py.

File: examples/vision/focal_modulation_network.py (hunk 4 above)
Comment (LakshmiKalaKadali) [reply examples/vision/focal_modulation_network.py]: Done

File: examples/vision/focal_modulation_network.py (hunk 5 above)
Comment (LakshmiKalaKadali) [reply examples/vision/focal_modulation_network.py:117]: Done
